Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -325,8 +325,7 @@ def _get_self_deployed_token_auth_code(self, refresh_token: str = '', resource:
 
         access_token = response_json.get('access_token', '')
         expires_in = int(response_json.get('expires_in', 3595))
-        if self.multi_resource:
-            refresh_token = response_json.get('refresh_token', '')
+        refresh_token = response_json.get('refresh_token', '')
```

Comment:
@dantavori I've removed this so it will take the refresh token even if `multi_resource` is not given - which is my case here.

Reply:
The refresh token initiated in line 308 is sufficient. Before I added the multi-resource logic we used the same refresh token and didn't update it (however, in multi-resource flow it's not possible because you make a multiple authorization calls for each resource and you must use only the last refresh token). basically it should work without the condition but maybe there is a reason for this logic that I'm not aware of... @Itay4 wdyt?

Reply:
i think we should always take the latest refresh token